### PR TITLE
Add Missing case compatibility test and definitions

### PR DESCRIPTION
**Pull request by GustavoCaso**
Add Missing Tests and definitions  for case compatibility.

I'm doubting about adding to `Form` and `Json` what do you think?

### CODE REVIEW

**Comment by GustavoCaso** [comment]
@solnic any update on this one?

File: spec/dry/types/array_spec.rb [hunk 1]
```diff
@@ -69,4 +69,26 @@
...
+
+  describe '#===' do
+    subject(:array) { Dry::Types['strict.array'].of(Dry::Types['strict.string']) }
+
+    it 'returns boolean' do
+      expect(array.===(%w(hello world))).to eq true
+      expect(array.===(['hello', 1234])).to eq false
+    end
+
+    context 'In case statement' do
```
Comment (solnic) [inline spec/dry/types/array_spec.rb]: `In` => `in`

File: spec/dry/types/array_spec.rb [hunk 2]
```diff
@@ -69,4 +69,26 @@
       expect(array.valid?(['five'])).to be(true)
     end
   end
+
+  describe '#===' do
+    subject(:array) { Dry::Types['strict.array'].of(Dry::Types['strict.string']) }
+
+    it 'returns boolean' do
+      expect(array.===(%w(hello world))).to eq true
+      expect(array.===(['hello', 1234])).to eq false
```
Comment (solnic) [inline spec/dry/types/array_spec.rb]: `eql true` => `eql(true)`

File: spec/dry/types/sum_spec.rb [hunk 3]
```diff
@@ -142,6 +142,28 @@
...
+
+    context 'In case statement' do
+      let(:value) do
+        case 'world'
+        when type then 'accepted'
+          else 'invalid'
+        end
+      end
+
+      it 'will return correct value' do
```
Comment (solnic) [inline spec/dry/types/sum_spec.rb]: `will return` => `returns`

**Comment by solnic** [comment]
In the future, please rebase and squash clean-up commits. It's easier to read history with fewer commits and it's useful when writing release notes / updating changelogs.

**Comment by flash-gordon** [comment]
@solnic fortunately, those are very rare, at least by me. FWIW, it'd be better to tag them with `[cleanup]`, or `[style]`, or something. From my experience, having huge commits with style changes makes them annoying for later digging through the history with tools like blame.

**Pull request merged**